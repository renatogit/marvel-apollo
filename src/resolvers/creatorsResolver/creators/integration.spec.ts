import CreatorsResolver from '.';
import DataSourcesCreatorsAPI from '@/dataSources/creators';
import TestServer from '@/configs/test';
import QUERY_GET_CREATORS from '@query/creators/queryCreators';
import MOCK_GET_CREATORS from '@mock/creators/mockCreators';

describe('resolvers/creatorsResolver/integration', () => {
	it('fetches creators and generate snapshots', async () => {
		const args = {
			dataSources: DataSourcesCreatorsAPI,
			resolver: CreatorsResolver.Query.creators,
			mock: MOCK_GET_CREATORS,
			variables: null,
			entity: 'creators',
			typeQuery: 'creators',
		};

		const {server} = await TestServer(args);

		const {body} = await server.executeOperation({
			query: QUERY_GET_CREATORS,
		});

		expect(body).toMatchSnapshot();
	});
});
