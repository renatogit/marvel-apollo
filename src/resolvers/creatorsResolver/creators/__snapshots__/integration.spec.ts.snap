// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/creatorsResolver/integration fetches creators and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "creators": {
        "attributionHTML": "mock data",
        "attributionText": "mock data",
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 1,
          "limit": 10,
          "offset": 0,
          "results": [
            {
              "fullName": "mock data",
              "id": 13970,
              "modified": "mock data",
              "suffix": "mock data",
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
          ],
          "total": 1,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
